app/controllers/course/assessment/question/programming_controller.rb

```diff
@@ -26,12 +30,19 @@ def edit
   end
 
   def update
-    @programming_question.assign_attributes programming_question_params
-    @programming_question.skills.clear if programming_question_params[:skill_ids].blank?
-    process_package
+    result = @programming_question.class.transaction do
+      @programming_question.assign_attributes programming_question_params
+      @programming_question.skills.clear if programming_question_params[:skill_ids].blank?
+      process_package
+      @programming_question.save
```

Comment:
`raise ActiveRecord::Rollback unless @programming_question.save`, otherwise `save` will return false and silently fail.